include/fastdds/rtps/writer/ReaderProxy.h

```diff
@@ -471,11 +448,13 @@ class ReaderProxy
      * Converts all changes with a given status to a different status.
      * @param previous Status to change.
      * @param next Status to adopt.
+     * @param func Function executed for each change which changes its status.
      * @return the number of changes that have been modified.
      */
     uint32_t convert_status_on_all_changes(
             ChangeForReaderStatus_t previous,
-            ChangeForReaderStatus_t next);
+            ChangeForReaderStatus_t next,
+            std::function<void(ChangeForReader_t& change)> func = {});
```

Comment:
Passing in an `std::function` by value produces dynamic allocations. You could pass a const reference, or make this a template method

Reply:
Done